lib/spack/spack/util/environment.py

```diff
@@ -338,16 +338,16 @@ def group_by_name(self):
             modifications[item.name].append(item)
         return modifications
 
-    def variable_is_possibly_set(self, var_name):
+    def variable_is_explicitly_unset(self, var_name):
```

Comment:
Minor note: I would shorten the name to `is_unset`. I think it's still self-documenting at that point, and I think that given our strict 79 character line limit we need to keep our helper function names as short as possible. I'd be fine with `variable_is_unset` also, but "explicitly" is definitely unnecessarily verbose. If we guarantee that it's unset, we guarantee that it's unset, I don't think we care how explicitly we did so.